guest/internal/prefilter/types.go

```diff
@@ -231,3 +232,10 @@ func (n *nodeInfo) ImageStates() map[string]*api.ImageStateSummary {
 
 	return n.imageStates
 }
+
+func (n *nodeInfo) UsedPorts() api.HostPortInfo {
+	if n.usedPorts != nil {
+		return n.usedPorts
+	}
+	return nil
```

Comment:
So, currently I don't see any parts actually fetching this `usedPorts` from host. Can you open another PR for this `usedPorts` support?

Reply:
You can see how other fields are fetched, e.g., `imageStates` above.